pkg/alertmanager/alertmanager.go

```diff
@@ -35,31 +35,33 @@ type Config struct {
 	// Used to persist notification logs and silences on disk.
 	DataDir     string
 	Logger      log.Logger
-	MeshRouter  gossipRouter
 	Retention   time.Duration
 	ExternalURL *url.URL
 }
 
 // An Alertmanager manages the alerts for one user.
 type Alertmanager struct {
-	cfg        *Config
-	api        *api.API
-	logger     log.Logger
-	nflog      nflog.Log
-	silences   *silence.Silences
-	marker     types.Marker
-	alerts     *mem.Alerts
-	dispatcher *dispatch.Dispatcher
-	inhibitor  *inhibit.Inhibitor
-	stop       chan struct{}
-	wg         sync.WaitGroup
-	router     *route.Router
+	cfg         *Config
+	api         *api.API
+	logger      log.Logger
+	nflog       *nflog.Log
+	silences    *silence.Silences
+	marker      types.Marker
+	alerts      *mem.Alerts
+	dispatcher  *dispatch.Dispatcher
+	inhibitor   *inhibit.Inhibitor
+	stop        chan struct{}
+	wg          sync.WaitGroup
+	router      *route.Router
+	peer        *cluster.Peer
+	peerTimeout time.Duration
 }
 
 // New creates a new Alertmanager.
-func New(cfg *Config) (*Alertmanager, error) {
+func New(peer *cluster.Peer, peerTimeout time.Duration, cfg *Config) (*Alertmanager, error) {
 	am := &Alertmanager{
 		cfg:    cfg,
+		peer:   peer,
```

Comment:
`peerTimeout` needs to be passed around too

Reply:
Good catch!  Though I confess I have no idea what it is used for, or what `clusterWait()` achieves.